From 4990992b51ee1d9132ce0926cefe6e3947f08e2e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:13 -0400
Subject: [tools] perf/tests: Factor make install tests

Message-id: <1408953210-26343-186-git-send-email-jolsa@redhat.com>
Patchwork-id: 88908
O-Subject: [PATCH RHEL7.1 BZ1133083 185/322] perf tests: Factor make install tests
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit ee4ad93e82b8fd260ae618f6b0413cd321bc09cb
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Dec 19 14:41:59 2013 +0100

description
===========
Factoring make install tests to check for multiple files. Adding default
set of installed files for install and install_bin tests.

Putting the 'test' line into the log file instead to the screen as it
gets more complex now.

If the tests fails to find a file, following message is displayed:

$ make -f tests/make make_install_bin
- make_install_bin: cd . && make -f Makefile DESTDIR=/tmp/tmp.nCVuQoSHaJ install-bin
  failed to find: bin/perf
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/make b/tools/perf/tests/make
index 2ca0abf..f641c35 100644
--- a/tools/perf/tests/make
+++ b/tools/perf/tests/make
@@ -106,10 +106,36 @@ test_make_python_perf_so := test -f $(PERF)/python/perf.so
 test_make_perf_o     := test -f $(PERF)/perf.o
 test_make_util_map_o := test -f $(PERF)/util/map.o
 
-test_make_install       := test -x $$TMP_DEST/bin/perf
-test_make_install_O     := $(test_make_install)
-test_make_install_bin   := $(test_make_install)
-test_make_install_bin_O := $(test_make_install)
+define test_dest_files
+  for file in $(1); do    \
+    if [ ! -x $$TMP_DEST/$$file ]; then  \
+      echo "  failed to find: $$file";  \
+    fi      \
+  done
+endef
+
+installed_files_bin := bin/perf
+installed_files_bin += etc/bash_completion.d/perf
+installed_files_bin += libexec/perf-core/perf-archive
+
+installed_files_plugins := lib64/traceevent/plugins/plugin_cfg80211.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_scsi.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_xen.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_function.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_sched_switch.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_mac80211.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_kvm.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_kmem.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_hrtimer.so
+installed_files_plugins += lib64/traceevent/plugins/plugin_jbd2.so
+
+installed_files_all := $(installed_files_bin)
+installed_files_all += $(installed_files_plugins)
+
+test_make_install       := $(call test_dest_files,$(installed_files_all))
+test_make_install_O     := $(call test_dest_files,$(installed_files_all))
+test_make_install_bin   := $(call test_dest_files,$(installed_files_bin))
+test_make_install_bin_O := $(call test_dest_files,$(installed_files_bin))
 
 # FIXME nothing gets installed
 test_make_install_man    := test -f $$TMP_DEST/share/man/man1/perf.1
@@ -162,7 +188,7 @@ $(run):
  cmd="cd $(PERF) && make -f $(MK) DESTDIR=$$TMP_DEST $($@)"; \
  echo "- $@: $$cmd" && echo $$cmd > $@ && \
  ( eval $$cmd ) >> $@ 2>&1; \
- echo "  test: $(call test,$@)"; \
+ echo "  test: $(call test,$@)" >> $@ 2>&1; \
  $(call test,$@) && \
  rm -f $@ \
  rm -rf $$TMP_DEST
@@ -174,7 +200,7 @@ $(run_O):
  cmd="cd $(PERF) && make -f $(MK) O=$$TMP_O DESTDIR=$$TMP_DEST $($(patsubst %_O,%,$@))"; \
  echo "- $@: $$cmd" && echo $$cmd > $@ && \
  ( eval $$cmd ) >> $@ 2>&1 && \
- echo "  test: $(call test_O,$@)"; \
+ echo "  test: $(call test_O,$@)" >> $@ 2>&1; \
  $(call test_O,$@) && \
  rm -f $@ && \
  rm -rf $$TMP_O \
-- 
1.7.1